### PR TITLE
Support LINGUAS for gnome.yelp() languages

### PR DESCRIPTION
**Pull request by ebassi**
Listing all languages inside meson.build for the Yelp-based manual
localisation is error-prone, and it also requires parsing and modifying
Meson files from external tools.

Just like we do for i18n, we can use an ancillary `LINGUAS` file in the
help source sub-directory to list all the help languages.

### CODE REVIEW

**Comment by ebassi** [comment]
The PR uses the `LINGUAS` file as a replacement for empty `languages` argument; I'm also open to either merge the two arrays, or to deprecate the `languages` argument in favour of only the `LINGUAS` file.

File: mesonbuild/scripts/yelphelper.py [hunk 1]
```diff
@@ -29,6 +29,22 @@
 parser.add_argument('--langs', dest='langs', default='')
 parser.add_argument('--symlinks', type=bool, dest='symlinks', default=False)
 
+def read_linguas(src_sub):
```
Comment (TingPing) [inline mesonbuild/scripts/yelphelper.py]: Probably a better way to share this with `gettext.py` than copy pasting it.

Comment (ebassi) [inline mesonbuild/scripts/yelphelper.py]: Would `from .gettext import read_linguas` be enough? Or should I move it to `__init__.py` and import it in both gettxt.py and yelphelper.py?

Comment (TingPing) [inline mesonbuild/scripts/yelphelper.py]: Leaving it `gettext.py` should be fine.

**Comment by jpakkane** [comment]
I'm all for the latter. There should be one, and preferably only one, obvious way to do it.

**Comment by ebassi** [comment]
@jpakkane Is it enough to remove the `languages` key from the `@permittedKwargs` decorators, in order to mark a key as deprecated?

**Comment by nirbheek** [comment]
I think we need a separate mechanism for explicit deprecations which will print a message telling the user what the replacement is.

**Comment by ebassi** [comment]
@nirbheek Looking at other methods in the gnome module, I see that deprecated named arguments are silently handled — like `namespace` in `gnome.gtkdoc()`. Maybe adding a new argument to the `@permittedKwargs` decorator for deprecated arguments, e.g.

```python
    @permittedKwargs({ 'foo', 'bar', 'baz', 'newbaz' }, deprecated={ 'baz': 'newbaz' })
    def somefunc(...):
        ...
```

**Comment by jpakkane** [comment]
Is this ready or should we do the deprecated kwarg one firsts?

**Comment by ebassi** [comment]
@jpakkane The existing PR only uses a `LINGUAS` file if the `langs` argument is not passed. I can amend it to at least warn about the deprecation. Then we can think about #2270.

**Pull request merged**